### PR TITLE
Personal Identifier Requirement List

### PR DESCRIPTION
**Pull request by nathaliemomeni**
# Security
Blocking Customer personal info in Reference Architecture.

## Why

It's important to TELUS that a TELUS customer’s Personal Identifiers are protected from TELUS users and 3rd party platform providers, so that we can further build on our brand's trust and reliability. We are committed to respecting our customers’ privacy and safeguarding their information. In a time where the threat to our industry continues to increase in various forms of fraudulent behaviour and cybercrime, the emphasis on a customer's privacy and security also increases.

For more clarity, for any development work that TELUS  customers' Personal Identifiers are accessed, developers need to block these personal info via snippets in TELUS pages.

 
## What

TELUS Privacy requirement from all pages and platforms that have access to TELUS  customers' Personal Information. TELUS customers' Personal information cannot be viewed, traced, or stored by TELUS employees or 3rd-party platform provider. 

The goal is to mitigate business risks and provide customer solutions that take every opportunity to protect:
- Customer information that is entrusted to us
- The TELUS brand and customer trust

## How
To protect customer privacy, TELUS  Privacy team requires all these Personal Identifiers to be blocked in all Reference Architecture pages at all times.

To mask these PI information using masking snippet, either provided by 3rd-party provider, or creating masking class internally at TELUS.

### TELUS PI-blocking Requirement:

- Customer  First Name
- Customer Last Name
- Email   Address
- Gender(title   Mr., Ms, etc)
- Home   Phone number
- Mobile   Phone Number
- Payment   Card # (Credit Card, Debit)
- Driver’s   License
- BAN
- Delinquency status
- Balance   Amount
- Transaction History
- Mailing address
- Billing Address
- Password
- PIN
- Chat session/transcript
- Usage details (numbers called/texted, how much usage),
- Preauth banking
- Security Question in Account profile section
- Security Answer in Account profile section
- Number of users in the Account

## Who

@Everyone

## References
A templated code is recommended to apply to all existing/new pages, to block these PI.

An example of masking class created by My TELUS developers:

https://github.com/telus/my-account-overview/blob/master/ui/src/components/Common/HideFromClicktale/HideFromClicktale.jsx

### CODE REVIEW

**Comment by ahmadnassri** [comment]
please use proper markdown for content in this wiki. attachments are not desired

**Comment by nathaliemomeni** [comment]
Sure, I removed the Excel sheet and provided the list there directly. Also, I followed the markdown that was in other pages in this repository.

**Comment by ahmadnassri** [comment]
@nathaliemomeni the content you wrote are just in the PR description, it needs to be a file in the wiki.

**Comment by nathaliemomeni** [comment]
Hi Ahmad,
Sorry what do you mean by PR?

I already have added the file in wiki: https://telusdigital.atlassian.net/wiki/spaces/AI/pages/522125464

**Comment by ahmadnassri** [comment]
@nathaliemomeni please sync up with @kspaans on this, also lets not duplicate content in two places, and please be aware this repo is *public* and viewable by the entire planet :)

**Comment by gonzalovazquez** [comment]
@ahmadnassri and @kspaans, we have updated the PR and removed the file binary.

**Comment by nathaliemomeni** [comment]
Thanks for your help @gonzalovazquez :)

File: security/pi.md [hunk 1]
```diff
@@ -2,50 +2,57 @@
...
+
+The goal is to mitigate business risks and provide customer solutions that take every opportunity to protect:
+
+- Customer information that is entrusted to us
+- The TELUS brand and customer trust
 
 ## How
 
-### Definition of PI:
+To protect customer privacy, Telus Privacy team requires all these Personal Identifiers to be blocked in all Reference Architecture pages at all times.
```
Comment (kspaans) [inline security/pi.md]: `Telus` again.

Also, is this referring to the pages that the customer sees, or the infrastructure that hosts those pages? We still need to, for example, display a customer's address on some of the My Account pages.

File: security/pi.md [hunk 2]
```diff
@@ -2,50 +2,57 @@
...
 
-Personal Information is any and all information about an identifiable individual. Personal information includes publicly available information, which can be found in a number of ways such as through social media sites and telephone directories. 
+To mask these PI information using masking snippet, either provided by 3rd-party provider, or creating masking class internally at Telus.
```
Comment (kspaans) [inline security/pi.md]: Is there an example of a masking snippet? I think this is referring to masking of data from back-end logging, but we could be more clear here.

Also, `Telus` again.

File: security/pi.md [hunk 3]
```diff
@@ -2,50 +2,57 @@
...
-- Marital status
-- Account number
-- SIN
-- Driver’s license
-- Pictures
-- Voice recordings
-- Education history
-- Opinions
-- Ethnic origin
+### TELUS PI-blocking Requirement:
```
Comment (ahmadnassri) [inline security/pi.md]: this list is significantly different than the one that was present before, are we sure we're not missing out anything here? e.g. `Pictures`, `Account Number`, `Marital Status` etc ...

Comment (nathaliemomeni) [reply security/pi.md]: @ahmadnassri this requirement list didnt exist, or atleast communicated by Security team before, and based on their recent validation, this is all the requirements they need TELUS teams to comply with. Also, Customer's picture, or Marital status is never asked in My Account pages/online properties, or anywhere in TELUS, to PI-block it.

File: security/pi.md (hunk 1 above)
Comment (nathaliemomeni) [reply security/pi.md]: Both @kspaans . These customer info is currently present in My Account pages, and technically Telus employees can see it, even if they dont do anything with it. With these PI-blocking, it wont be visible to TELUS employees either.
Also, in 3rd party platforms, if PI-blocking scripts would be applied, no 3rd party platform can capture/see/store customer info.

File: security/pi.md [hunk 4]
```diff
@@ -2,50 +2,57 @@
 
 ## Why
 
-Special care needs to be given to handling of PI - Personal Information - in order to comply with [TELUS privacy and security policy](https://www.telus.com/en/bc/support/privacy-policy).
+It's important to Telus that a TELUS customer's Personal Identifiers are protected from Telus users and 3rd party platform providers, so that we can further build on our brand's trust and reliability. We are committed to respecting our [customers' privacy](https://www.telus.com/en/bc/support/privacy-policy) and safeguarding their information. In a time where the threat to our industry continues to increase in various forms of fraudulent behaviour and cybercrime, the emphasis on a customer's privacy and security also increases.
```
[the remark this answers is not in the document]
Comment (nathaliemomeni) [reply security/pi.md]: Sorry, sure will correct it.

File: security/pi.md (hunk 2 above)
Comment (nathaliemomeni) [reply security/pi.md]: @kspaans yes Ive put example of a masking snippet that My Telus developers has created for their own purposes. But considering Github is a public platform, we may actually have to remove this reference: A templated code is recommended to apply to all existing/new pages, to block these PI.

An example of masking class created by My TELUS developers:

https://github.com/telus/my-account-overview/blob/master/ui/src/components/Common/HideFromClicktale/HideFromClicktale.jsx

File: security/pi.md (hunk 1 above)
Comment (kspaans) [reply security/pi.md]: OK, can we clarify the language here? This wording looks like it's saying that PI shouldn't be shown to customers on the My TELUS pages.

Comment (nathaliemomeni) [reply security/pi.md]: Sure @kspaans , I revised it. Is it more clear now? 
To protect customer privacy, TELUS Privacy team requires all these Personal Identifiers entered by TELUS customer, to be blocked to TELUS employees and vendors, in all Reference Architecture pages at all times. Personal Identifiers, as entered by TELUS customers, should only be visible to TELUS customer, and no one else.

**Comment by WITMiddle** [comment]
Submitted a commit to change "Reference Architecture" with "all TELUS github repositories" to reflect the proper intentions. "Reference Architecture" can be misleading as it refers to Reference Architecture Wiki repo only. (which is a public repo where no PI or sensitive data of any kind should be present)

**Comment by kspaans** [comment]
I'm not sure if the purpose of this page is to identify all of the types of PI, or to do that and also identify where we should avoid storing or showing PI. I think it is complete for the first part, but I'm not sure about the last part. It mentions masking, but does not, for example, say to avoid storing any PI in logs, or exposing it through URL parameters.

I think we could make this more general and simply say that PI should never be exposed to anyone but the customer who owns it, and should never be stored on any non-TELUS systems or services. The technical details can be explored on another page, or left up to the discretion of the developers working with systems that touch PI.

We can also start coordinating around a shared project for PI masking, as the AI team (via @crimeminister) has a start on this: https://github.com/telus/dlp-types

**Comment by nathaliemomeni** [comment]
@kspaans , it doesnt matter if it would be in this PR, or AI team's PR, but we need to have one reference point for all Telus POs and devs to follow, to block PI and meet Security team's requirement.
The goal of this PR was to create such a reference for Telus devs.

Whichever repo we do it at, we should provide ALL the info(from Security perspective) to Developers in one place that would be relevant/updated for foreseeable future, to avoid confusion and hit and miss.

**Comment by kspaans** [comment]
@nathaliemomeni OK cool, I think this accomplishes that then! :+1:

**Comment by nathaliemomeni** [comment]
Thanks for approving @kspaans :)

**Comment by nathaliemomeni** [comment]
Hi @kspaans , Emil Barbon from MyTelus team made a case about these PI-blocking, and Elena Novas and Cameron Sawatzky revised their decision on PI-blocking requirement above.  I'll edit the original version to reflect it.

**Pull request merged**